spinnaker/spinnaker_system/google_http_lb_upsert_test.py

```diff
@@ -0,0 +1,894 @@
+    hc_clause_builder = hc_clause_builder.list_resource('httpHealthChecks')
+    for hc in health_checks:
+      hc_clause_builder.contains_pred_list(
+        [
+          jp.PathEqPredicate('name', hc['name']),
+          jp.PathEqPredicate('requestPath', hc['requestPath']),
+          jp.PathEqPredicate('port', hc['port'])
+        ]
+      )
+
+    bs_clause_builder = contract_builder.new_clause_builder('Backend Services Created',
+                                                            retryable_for_secs=30)
+    bs_clause_builder = bs_clause_builder.list_resource('backendServices')
+    for bs in backend_services:
+      bs_clause_builder.contains_pred_list(
+        [
+         jp.PathEqPredicate('name', bs['name']),
+         jp.PathEqPredicate('portName', 'http'),
+         jp.PathContainsPredicate('healthChecks[0]', bs['healthCheck']['name'])
+        ]
+      )
+
+    url_map_clause_builder = contract_builder.new_clause_builder('Url Map Created',
+                                                                 retryable_for_secs=30)
+    url_map_clause_builder = url_map_clause_builder.list_resource('urlMaps')
+    for hr in host_rules:
+      pm = hr['pathMatcher']
+      pred_list = [
+         jp.PathEqPredicate('name', self.__lb_name),
+         jp.PathEqPredicate('defaultService',
+                                  self._make_bs_link(upsert['defaultService']['name'])),
+         jp.PathContainsPredicate('pathMatchers/defaultService',
+                                  self._make_bs_link(pm['defaultService']['name'])),
+      ]
+      for host in hr['hostPatterns']:
+        pred_list.append(
+         jp.PathContainsPredicate('hostRules/hosts', host),
+        )
+      for pr in pm['pathRules']:
+        pred_list.append(
+         jp.PathContainsPredicate('pathMatchers/pathRules/service',
+                                  self._make_bs_link(pr['backendService']['name'])),
+        )
+        for path in pr['paths']:
+          pred_list.append(
+            jp.PathContainsPredicate('pathMatchers/pathRules/paths', path),
+          )
+      url_map_clause_builder.contains_pred_list(
+        pred_list
+      )
+
+    fr_clause_builder = contract_builder.new_clause_builder('Forwarding Rule Created',
+                                                            retryable_for_secs=30)
+    fr_clause_builder = fr_clause_builder.list_resource('globalForwardingRules')
+
+    port_string = '443-443'
+    if upsert['certificate'] == '':
+      port_string = '%s-%s' % (upsert['portRange'], upsert['portRange'])
+
+    fr_clause_builder.contains_pred_list(
+      [
+         jp.PathEqPredicate('name', self.__lb_name),
+         jp.PathEqPredicate('portRange', port_string)
+      ]
+    )
+
+    proxy_clause_builder = contract_builder.new_clause_builder('Target Proxy Created',
+                                                               retryable_for_secs=30)
+    if upsert['certificate'] == '':
```

Comment:
actually, both clauses are the same except for a couple parameters. I'd either create a parameterized function or define the variables here.

if upsert['certificate']:
  makeTargetProxyCreatedClause('targetHttpProxies', HTTP_PROXY_PREFIX)
else:
  makeTargetProxyCreatedClause('targetHttpsProxies', HTTPS_PROXY_PREFIX)

but since *PROXY_PREFIX isnt actually referenced anywhere, this is just:
  kind = 'Https' if clause else 'Http'
  target_proxy_name = '%s-target-%s-proxy' % (self.__lb_name, kind.lower())
  contract_builder.new_clause_builder('Target Proxy Created', retryable_for_secs=30)
    .list_resource('target{0}Proxies'.format(kind))
    .contains_path_eq('name', target_proxy_name)